enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/components/Filter.tsx

```diff
@@ -24,10 +24,10 @@ const FilterInner = ({
 }: {
   question: Question;
 } & FilterProps) => {
-  const { onQuestionChange } = useInteractiveQuestionContext();
+  const { updateQuestion } = useInteractiveQuestionContext();
 
   const onQueryChange = (query: Lib.Query) =>
-    onQuestionChange(question.setQuery(query));
+    updateQuestion(question.setQuery(query), { run: true });
```

Comment:
If I understand correctly, `onQueryChange` is being called AFTER clicking on "Apply Filters", so it makes sense to attempt auto-run here.